lambdas/ec2/README.md

```diff
@@ -0,0 +1,39 @@
+# AWS EC2 Reaper 
+
+This AWS EC2 Reaper works using tags set on the instance itself. The Reaper is composed of two AWS Lambdas: the "Schema Enforcer" and the "Terminator". The Schema Enforcer ensures that all instances have a valid `termination_date` tag set when the EC2 resource is created, and the Terminator runs periodically to delete all instances past their `termination_date`.
+
+## Usage
+AWS EC2 resources should have a `lifetime` tag added during resource creation. A valid value is a string of of an integer value with a 1 letter unit of w(weeks), d(days), h(hours). For example, `1w` is 1 week, `2d` is 2 days, and `3h` is 3 hours. The "Schema Enforcer" will convert that `lifetime` tag to a `termination_date`. Users can bypass this workflow with the `lifetime` key and set the `termination_date` directly; the `termination_date` must be a valid IS0 8601 value with a UTC offset defined. If no `termination_date` is found within four minutes of creation, the resource is terminated. 
+
+If the lifetime for an instance needs to be prolonged, the `termination_date` should be updated directly.
+
+## Installation 
+Currently, there is no build script yet for the Reaper; you will need to copy the `schema_enforcer.py` and `terminator.py` files to Lambdas in AWS. You will also need to ensure that there is a role for the lambda's with sufficient privilege and access to read events and delete instances. Once those are in place, you will need to add a rule in Cloudwatch for the Enforcer with this event pattern:
+
+```json
+{
+  "source": [
+    "aws.ec2"
+  ],
+  "detail-type": [
+    "EC2 Instance State-change Notification"
+  ],
+  "detail": {
+    "state": [
+      "pending"
+    ]
+  }
+}
+```
+
+The Terminator should run on a Cloudwatch schedule, configurable in the AWS EC2 GUI. 
+
+## Components
+
+### Schema Enforcer
+The Schema Enforcer is a lambda that is designed to be triggered when an EC2 instance enters the 'pending' state. This lambda waits for an ec2 instance to have a valid `lifetime` tag associated with it. When that tag is found, it converts that tag to a UTC date and adds a `termination_date` tag to the instance with that UTC date as its value. The Schema Enforcer terminates instances that do no have valid `termination_date` tags after 4 minutes, and reports that termination as an exception raised. The Schema Enforcer does not terminate instances past their `termination_date`; the Terminator is responsible for that.
```

Comment:
I think this description is missing a portion that is in the "Usage" section above:

>  Users can bypass this workflow with the `lifetime` key and set the `termination_date` directly

If a host doesn't have a `lifetime` tag, the schema enforcer will never process that tag and convert it into a 'termination_date' tag. However, you can supply a `termination_date` tag directly, and it will just process that `termination_date` tag without the lifetime conversion.